model/group.go

```diff
@@ -157,6 +159,20 @@ func (group *Group) IsValidForUpdate() *AppError {
 	return nil
 }
 
+var validGroupnameChars = regexp.MustCompile(`^[a-z0-9\.\-_]+$`)
+
+func (group *Group) IsValidName() *AppError {
+	if l := len(group.Name); l == 0 || l > GroupNameMaxLength {
+		return NewAppError("Group.IsValidName", "model.group.name.app_error", map[string]interface{}{"GroupNameMaxLength": GroupNameMaxLength}, "", http.StatusBadRequest)
+	}
+
+	if !validGroupnameChars.MatchString(group.Name) {
+		return NewAppError("Group.IsValidName", "model.group.name.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Add more detail to this error. Such as "Contains invalid characters" or a separate error.

Reply:
added separate error.